### PR TITLE
Coupla fixes

### LINKED ISSUE #585: apply constraint.subinplace to substitutions
As it stands, substitutions may get stranded during a varkey/varkey substitution.

### LINKED ISSUE #782: GPs should not accept varkeys/monomials as values in their substitutions

### PR DESCRIPTION
**Pull request by bqpd**
@whoburg, ready for review

Closes #585 
Closes #782

### CODE REVIEW

**Comment by bqpd** [comment]
test models please

File: .gitignore [hunk 1]
```diff
@@ -57,6 +57,7 @@ coverage.xml
 
 # Sphinx documentation
 docs/_build/
+docs/source/autodoc/gpkit.rst-e
```
Comment (whoburg) [inline .gitignore]: @bqpd please review this commit before merging

File: gpkit/constraints/set.py [hunk 2]
```diff
@@ -168,6 +168,8 @@ def subinplace(self, subs):
         "Substitutes in place."
         for constraint in self:
             constraint.subinplace(subs)
+        for key in set(self.substitutions).intersection(subs):
+            self.substitutions[subs[key]] = self.substitutions[key]
```
Comment (whoburg) [inline gpkit/constraints/set.py]: I could be wrong here, but something seems fishy here:

`subs[key]` is a value.
... and now we're using that value as a key into `self.substitutions`?

just making sure.

File: gpkit/tests/t_sub.py [hunk 3]
```diff
@@ -12,6 +12,16 @@
...
+    def test_bad_gp_sub(self):
+        x = Variable("x")
+        y = Variable("y")
+        m = Model(x, [y >= 1], {y: x})
+        with self.assertRaises(ValueError):
+            m.solve()
+        m = Model(x, [y >= 1])
+        m.subinplace({y: x})
+        self.assertAlmostEqual(m.solve(verbosity=0)["cost"], 1)
+
```
Comment (whoburg) [inline gpkit/tests/t_sub.py]: nice test

**Comment by whoburg** [comment]
test models please

File: gpkit/constraints/set.py (hunk 2 above)
Comment (bqpd) [inline gpkit/constraints/set.py]: I forgot to check that `subs[key]` is a string, varkey, or monomial, my bad!

File: .gitignore (hunk 1 above)
Comment (bqpd) [inline .gitignore]: Err, what's this do?

Comment (bqpd) [inline .gitignore]: That's not an autogenerated file, unlike the rest in that folder...

File: gpkit/constraints/set.py (hunk 2 above)
Comment (bqpd) [inline gpkit/constraints/set.py]: Fixed.

**Comment by bqpd** [comment]
test models please

File: .gitignore (hunk 1 above)
Comment (bqpd) [inline .gitignore]: Document in-line?

Comment (bqpd) [inline .gitignore]: Removing as out of scope, but feel free to make another PR!

**Comment by bqpd** [comment]
To be returned to when Jenkins is back up.

**Comment by bqpd** [comment]
test gpkit please

**Comment by bqpd** [comment]
test models please

**Comment by bqpd** [comment]
test models please

**Pull request merged**